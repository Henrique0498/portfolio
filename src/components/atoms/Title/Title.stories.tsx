import { Meta, StoryObj } from '@storybook/react'

import Title from '.'

export default {
  title: 'Title',
  component: Title
} as Meta

export const Default: StoryObj = {
  args: {
    children: 'Title',
    className: 'text-3xl text-blue-600 dark:text-red-600'
  }
}
