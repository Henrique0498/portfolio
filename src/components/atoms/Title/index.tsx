import styles from './styles.module.scss'
import '@testing-library/jest-dom'

interface InTitle extends React.ComponentPropsWithRef<'h1'> {}

const Title = ({ children, className = '', ...props }: InTitle) => {
  return (
    <h1 {...props} className={`${styles.container} ${className}`}>
      {children}
    </h1>
  )
}
export default Title
