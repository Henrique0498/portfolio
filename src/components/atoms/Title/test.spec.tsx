import { render, screen } from '@testing-library/react'

import Title from '.'

describe('<Title />', () => {
  it('should render the children', () => {
    const { container } = render(<Title>Testando...</Title>)

    expect(
      screen.getByRole('heading', { name: /testando/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the font size correctly', () => {
    render(<Title>Testando...</Title>)
    const titleElement = screen.getByRole('heading', {
      name: /Testando\.\.\./i
    })

    expect(titleElement).toHaveClass('container')
  })
})
