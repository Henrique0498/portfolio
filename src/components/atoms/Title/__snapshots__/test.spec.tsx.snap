// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render the children 1`] = `
<h1
  class="container "
>
  Testando...
</h1>
`;
